scss/tests/mixins/_utilities-map.test.scss

```diff
@@ -0,0 +1,339 @@
+        @include before ();
+
+        @include utilities-add-values(align, (
+          middle: middle
+        ));
+
+        @include assert-equal(utilities-get-option(align, values), (
+          baseline: baseline,
+          top: top,
+          middle: middle
+        ));
+      }
+    }
+    @include describe("utilities with single value") {
+      @include it("Creates a map and merges the new values") {
+        @include before ();
+
+        @include utilities-add-values(margin, (
+          5pct: 5%
+        ));
+
+        @include assert-equal(utilities-get-option(margin, values), (
+          auto: auto,
+          5pct: 5%
+        ));
+      }
+    }
+  }
+
+  @include describe("utilities-remove-values") {
+    @include describe("utilities with map of values") {
+      @include it("removes the value with the given key") {
+        @include before ();
+
+        @include utilities-remove-values(flex, fill);
+
+        @include assert-equal(utilities-get-option(flex, values),());
+      }
+    }
+    @include describe("utilities with list of values") {
+      @include it("removes the value and creates a map") {
+        @include before ();
+
+        @include utilities-remove-values(align, top);
+
+        @include assert-equal(
+          utilities-get-option(align, values),
+          (
+            baseline: baseline
+          )
+        );
+      }
+    }
+    @include describe("utilities with single value") {
+      @include it("removes the value and creates a map") {
+        @include before ();
+
+        @include utilities-remove-values(margin, auto);
+
+        @include assert-equal(
+          utilities-get-option(margin, values),
+          ()
+        );
+      }
+    }
+    @include describe("$values...") {
+      @include it("allows to remove multiple values") {
+        @include before ();
+
+        @include utilities-remove-values(align, baseline, top);
+
+        @include assert-equal(utilities-get-option(align, values),());
+      }
+
+      @include it("ignores unknown values") {
+        @include before ();
+
+        @include utilities-remove-values(flex, unknown, fill, non-existing);
+
+        @include assert-equal(utilities-get-option(flex, values), ());
+      }
+    }
+  }
+
+  @include describe("utilities-add") {
+    @include it("adds the utility") {
+      @include before ();
+
+      @include utilities-add(z-index, (
+        class: z,
+        property: z-index,
+        values: (-1, 0, 1, 2, 3)
+      ));
+
+      @include assert-equal(utilities-get-options(z-index), (
+        class: z,
+        property: z-index,
+        values: (-1, 0, 1, 2, 3)
+      ));
+    }
+
+    @include it("overrides an existing utility") {
+      @include before ();
+
+      @include utilities-add("flex", (
+        class: flex,
+        values: auto none
+      ));
+
+      @include assert-equal(utilities-get-options(flex), (
+        class: flex,
+        values: auto none
+      ));
```

Comment:
Do we need to check that the previous one doesn't exist anymore ?

Reply:
The assertion compares that the two map of options are equal so will detect if the previous ones are still there. Had a quick double check to confirm, as it wasn't really clear from the documentations and the following does fail because of the extra value:

```scss
@include it("compares map completely") {
  @include assert-equal((
    value: 1,
    extra-value: 2
  ), (value: 1));
}
```

Reply:
Ok perfect, thanks a lot for checking :smile: